### PR TITLE
Crowdsale Contract Creation Form Layout

### PR DESCRIPTION
**Pull request by Kamikozz**

### CODE REVIEW

File: src/pages/CrowdsaleContract/CrowdsaleContract.helpers.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
+import React, { ReactElement } from 'react';
+import { TextFieldProps } from '@material-ui/core';
+import * as Yup from 'yup';
+import { Calendar, DescendingSortOrderIcon } from 'theme/icons';
+
+const latinAndNumbers = /^[A-Za-z][A-Za-z0-9][0-9A-Za-z]*$/;
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/CrowdsaleContract.helpers.tsx:6]: we have it in 2 files now, lets move it to constants to have one source

File: src/pages/CrowdsaleContract/CrowdsaleContract.helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    renderProps: {
+      label: 'Token rate',
+      name: 'tokenRate',
+      type: 'input',
+    },
+    helperText: [
+      'Defines the rate of your token. 1-100000 numerals are accepted.',
+    ],
+  },
+  // {
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/CrowdsaleContract.helpers.tsx:135]: delete comments

File: src/pages/CrowdsaleContract/CrowdsaleContract.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    width: '90%',
+    [theme.breakpoints.down('sm')]: {
+      width: '100%',
+    },
+  },
+  crowdsaleContractFormSection: {
+    borderTop: `1px solid ${theme.palette.type === 'dark' ? COLOR_BLACK_3 : COLOR_GREY_5}`,
+    padding: `${theme.spacing(5)}px 0px`,
+  },
+  shortInput: {
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/CrowdsaleContract.styles.tsx:20]: i think its a not working part of my code, which moved to grid to make some inputs shorter, delete

File: src/pages/CrowdsaleContract/CrowdsaleContract.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    background: theme.palette.type === 'dark' ? COLOR_BLACK_1 : COLOR_GREY_2,
+    borderRadius: theme.spacing(2.5),
+    marginBottom: theme.spacing(4),
+    [theme.breakpoints.down('sm')]: {
+      padding: `${theme.spacing(3)}px ${theme.spacing(2)}px`,
+    },
+  },
+
+  changingDatesHeader: {
+    ...flexHelper('space-between', 'center'),
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/CrowdsaleContract.styles.tsx:50]: center is default value, you can omit it

File: src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { COLOR_BLACK_4, COLOR_ACID_GREEN } from 'theme/colors';
+import { flexHelper } from 'utils';
+
+export const useStyles = makeStyles((theme: Theme) => createStyles({
+  root: {
+    padding: `${theme.spacing(4)}px ${theme.spacing(3)}px`,
+    // background: theme.palette.type === 'dark' ? COLOR_BLACK_1 : COLOR_GREY_2,
+    borderRadius: theme.spacing(1.5),
+    // marginBottom: theme.spacing(4),
+    border: `1px solid ${theme.palette.type === 'dark' ? COLOR_BLACK_4 : COLOR_BLACK_4}`,
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx:12]: i know there is now color for light theme in design, just check that similar colored blocks has the same border color or change it

File: src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { makeStyles } from '@material-ui/core';
+import { createStyles, Theme } from '@material-ui/core/styles';
+import { COLOR_BLACK_4, COLOR_ACID_GREEN } from 'theme/colors';
+import { flexHelper } from 'utils';
+
+export const useStyles = makeStyles((theme: Theme) => createStyles({
+  root: {
+    padding: `${theme.spacing(4)}px ${theme.spacing(3)}px`,
+    // background: theme.palette.type === 'dark' ? COLOR_BLACK_1 : COLOR_GREY_2,
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx:9]: delete comments

File: src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    borderRadius: theme.spacing(1.5),
+    // marginBottom: theme.spacing(4),
+    border: `1px solid ${theme.palette.type === 'dark' ? COLOR_BLACK_4 : COLOR_BLACK_4}`,
+    [theme.breakpoints.down('sm')]: {
+      padding: `${theme.spacing(3)}px ${theme.spacing(2)}px`,
+    },
+  },
+  title: {
+    ...flexHelper('flex-start', 'flex-start'),
+    gap: theme.spacing(1),
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx:19]: gap not working in safari, dont use it

File: src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    marginBottom: theme.spacing(2),
+    '& > svg': {
+      fill: theme.palette.type === 'dark' ? COLOR_ACID_GREEN : COLOR_ACID_GREEN,
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx:22]: remover redundant ternary expressions

File: src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    },
+  },
+  titleText: {
+    color: theme.palette.type === 'dark' ? COLOR_ACID_GREEN : COLOR_ACID_GREEN,
+  },
+  // deleteIcon: {
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx:28]: delete

File: src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.styles.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    padding: `${theme.spacing(4)}px ${theme.spacing(3)}px`,
+    background: theme.palette.type === 'dark' ? COLOR_BLACK_1 : COLOR_GREY_2,
+    borderRadius: theme.spacing(2.5),
+    marginBottom: theme.spacing(4),
+    [theme.breakpoints.down('sm')]: {
+      padding: `${theme.spacing(3)}px ${theme.spacing(2)}px`,
+    },
+  },
+  header: {
+    ...flexHelper('space-between', 'center'),
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.styles.tsx:17]: center is default, omit

File: src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.styles.tsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    marginBottom: theme.spacing(1),
+  },
+
+  title: {
+    ...flexHelper('flex-start', 'flex-start'),
+    color: theme.palette.type === 'dark' ? COLOR_GREY_1 : COLOR_BLACK_1,
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.styles.tsx:23]: dont use unnecessary ternary expressions

File: src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+          render={() => (
+            <Switch
+              name={checkboxName}
+              checked={checked}
+              onClick={onChecked}
+            />
+          )}
+        />
+      </Box>
+      { description && (
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.tsx:42]: use !! description

File: src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.tsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        <Typography
+          className={classes.description}
+          variant="body1"
+          color="textSecondary"
+        >
+          {description}
+        </Typography>
+      )}
+      <Grid container>
```
Comment (ValnetineKul) [inline src/pages/CrowdsaleContract/components/SwitchableBlockForm/SwitchableBlockForm.tsx:51]: i think its better not to use grid container here and use it with children, it creates unexpected logic not it this case but in some others

**Review by ValnetineKul — CHANGES_REQUESTED** [review]

File: src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx (hunk 7 above)
Comment (Kamikozz) [reply src/pages/CrowdsaleContract/components/InfoBlock/InfoBlock.styles.tsx:19]: now supported, but only for >=Safari 14.1

**Pull request merged**